.github/workflows/test.yml

```
@@ -0,0 +1,60 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  RUBY_VERSION: 3.0.2
+  NODE_VERSION: 16.9.1
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-20.04
+    services:
+      postgres:
+        image: postgres:11
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_HOST_AUTH_METHOD: trust
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ env.RUBY_VERSION }}
+          bundler-cache: true
+
+      - name: Bundle Install
+        run: bundle install
+
+      - name: Setup & create Database
+        run: bundle exec rake db:test:prepare
+        env:
+          RAILS_ENV: test
+          DATABASE_USERNAME: postgres
+          DATABASE_PASSWORD: postgres
+
+      - name: Precompile assets
+        run: |
+          npm install
+          bundle exec rake assets:precompile
+        env:
+          RAILS_ENV: test
```

You need to include these lines to make the CI work.

```suggestion
          RAILS_ENV: test
          DATABASE_USERNAME: postgres
          DATABASE_PASSWORD: postgres
```